diff --git a/node_modules/@cdktn/provider-generator/lib/get/generator/resource-parser.js b/node_modules/@cdktn/provider-generator/lib/get/generator/resource-parser.js
index cde9b9c..135f836 100644
--- a/node_modules/@cdktn/provider-generator/lib/get/generator/resource-parser.js
+++ b/node_modules/@cdktn/provider-generator/lib/get/generator/resource-parser.js
@@ -81,10 +81,10 @@ class Parser {
         this.structs = new Array();
     }
     uniqueClassName(className) {
-        if (this.classNames.includes(className)) {
+        if (this.classNames.includes(className.toLowerCase())) {
             className = `${className}A`;
         }
-        this.classNames.push(className);
+        this.classNames.push(className.toLowerCase());
         return className;
     }
     resourceFrom(fqpn, type, schema, terraformSchemaType, constraint) {
