diff --git a/node_modules/cdktn/lib/complex-computed-list.js b/node_modules/cdktn/lib/complex-computed-list.js
index ba7a323..08c0fcd 100644
--- a/node_modules/cdktn/lib/complex-computed-list.js
+++ b/node_modules/cdktn/lib/complex-computed-list.js
@@ -132,6 +132,23 @@ class AnyMap extends ComplexResolvable {
 exports.AnyMap = AnyMap;
 _d = JSII_RTTI_SYMBOL_1;
 AnyMap[_d] = { fqn: "cdktn.AnyMap", version: "0.22.1" };
+// eslint-disable-next-line jsdoc/require-jsdoc
+class StringMapMap extends ComplexResolvable {
+    constructor(terraformResource, terraformAttribute) {
+        super(terraformResource, terraformAttribute);
+        this.terraformResource = terraformResource;
+        this.terraformAttribute = terraformAttribute;
+    }
+    lookup(key) {
+        return new StringMap(this.terraformResource, `${this.terraformAttribute}["${key}"]`);
+    }
+    computeFqn() {
+        return tokens_1.Token.asString(this.terraformResource.interpolationForAttribute(this.terraformAttribute));
+    }
+}
+exports.StringMapMap = StringMapMap;
+_e = JSII_RTTI_SYMBOL_1;
+StringMapMap[_e] = { fqn: "cdktn.StringMapMap", version: "0.22.1" };
 /**
  * @deprecated Going to be replaced by Array of ComplexListItem
  * and will be removed in the future
